gen3-integration-tests/tests/test_indexing_page.py

```diff
@@ -0,0 +1,120 @@
+import os
+import utils.gen3_admin_tasks as gat
+
+from utils import logger
+from gen3.auth import Gen3Auth
+from gen3.index import Gen3Index
+from pages.login import LoginPage
+from pages.indexing_page import IndexingPage
+from services.indexd import Indexd
+
+
+@pytest.mark.portal
+@pytest.mark.sower
+@pytest.mark.ssjdispatcher
+class TestIndexingPage:
+    variables = {}
+
+    def setup_class(cls):
+        cls.variables["test_guid"] = "c2da639f-aa25-4c4d-8e89-02a143788268"
+        cls.variables["test_hash"] = (
+            "73d643ec3f4beb9020eef0beed440ad4"  # pragma: allowlist secret
+        )
+        cls.variables["expected_result"] = (
+            f"{cls.variables['test_guid']},"
+            "s3://cdis-presigned-url-test/testdata,,jenkins2,"
+            f"{cls.variables['test_hash']},13,"
+        )  # pragma: allowlist secret
+
+    def teardown_class(cls):
+        # Delete the indexd record after the test
+        user = "indexing_account"
+        auth = Gen3Auth(refresh_token=pytest.api_keys[user])
+        indexd = Gen3Index(auth_provider=auth)
+        try:
+            indexd.delete_record(guid=cls.variables["test_guid"])
+        except Exception as e:
+            logger.exception(
+                msg=f"Failed to delete record for guid {cls.test_guid}: {e}"
+            )
+
+    def test_indexing_upload_valid_manifest(self, page):
+        """
+        Scenario: Login and navigate to indexing page and upload dummy manifest
+        Steps:
+            1. Login with indexing_account user
+            2. Navigate to indexing page
+            3. Upload valid manifest and wait for manifest-indexing sower job to finish
+            4. Check if the indexd record exists
+            5. Check if the indexd record hashes exists
+        """
+        login_page = LoginPage()
+        indexing_page = IndexingPage()
+        indexd = Indexd()
+        # Go to login page and login with indexing_account user
+        login_page.go_to(page)
+        login_page.login(page, user="indexing_account")
+        # Go to indexing page and validate page is loaded
+        indexing_page.go_to(page)
+        # Upload the valid manifest via indexing page
+        indexing_page.upload_valid_indexing_manifest(page)
+        # Check if the sowerjob pod for manifest-indexing has completed
+        gat.check_pod(pytest.namespace, "manifest-indexing", "sowerjob")
+        # Get the indexd record and check if the hash value matches to the test_hash value
+        index_record = indexd.get_record(self.variables["test_guid"])
+        indexd_record_hash = index_record["hashes"]["md5"]
+        logger.info(indexd_record_hash)
+        assert (
+            indexd_record_hash == self.variables["test_hash"]
+        ), f"Expected MD5 hash {self.variables['test_hash']}, but got {indexd_record_hash}"
+
+    def test_indexing_download(self, page):
+        """
+        Scenario: Login and navigate to indexing page and download indexd manifest
+        Steps:
```

Comment:
it looks like this test depends on the previous test succeeding. if that's the case, can we merge them into a single end-to-end test?

Reply:
yes, we should combine them